src/main/java/com/mparticle/kits/BranchUtil.java

```diff
@@ -230,7 +166,14 @@ private void updateEventWithInfo(BranchEvent event, BranchUniversalObject buo, M
     // Region Translate CommerceEvents
 
     BranchEvent createBranchEventFromMPCommerceEvent(CommerceEvent event) {
-        BranchEvent branchEvent = createBranchEventFromEventName(event.getProductAction().toLowerCase());
+        BranchEvent branchEvent;
+        if (event.getProductAction() != null) {
+            branchEvent = createBranchEventFromEventName(event.getProductAction().toLowerCase());
+        } else if (event.getPromotionAction() != null) {
+            branchEvent = createBranchEventFromPromotionEventName(event.getPromotionAction().toLowerCase());
```

Comment:
it seems a little out of place to call `toLowerCase()` here, when you are going to look for the value populated with non-`toLowerCase()`'d keys. It works because all of out `Product` actions and `Promotion` actions are all lower case (and we are not changing those anytime soon), but I don't get why